shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm

```diff
@@ -0,0 +1,378 @@
+    case ui::AXEventGenerator::Event::RELATED_NODE_CHANGED:
+    case ui::AXEventGenerator::Event::REQUIRED_STATE_CHANGED:
+    case ui::AXEventGenerator::Event::ROLE_CHANGED:
+    case ui::AXEventGenerator::Event::SCROLL_HORIZONTAL_POSITION_CHANGED:
+    case ui::AXEventGenerator::Event::SCROLL_VERTICAL_POSITION_CHANGED:
+    case ui::AXEventGenerator::Event::SELECTED_CHANGED:
+    case ui::AXEventGenerator::Event::SET_SIZE_CHANGED:
+    case ui::AXEventGenerator::Event::SORT_CHANGED:
+    case ui::AXEventGenerator::Event::STATE_CHANGED:
+    case ui::AXEventGenerator::Event::TEXT_ATTRIBUTE_CHANGED:
+    case ui::AXEventGenerator::Event::VALUE_MAX_CHANGED:
+    case ui::AXEventGenerator::Event::VALUE_MIN_CHANGED:
+    case ui::AXEventGenerator::Event::VALUE_STEP_CHANGED:
+    case ui::AXEventGenerator::Event::WIN_IACCESSIBLE_STATE_CHANGED:
+      // There are some notifications that aren't meaningful on Mac.
+      // It's okay to skip them.
+      break;
+  }
+  return events;
+}
+
+void AccessibilityBridgeMacDelegate::DispatchAccessibilityAction(ui::AXNode::AXID target,
+                                                                 FlutterSemanticsAction action,
+                                                                 const std::vector<uint8_t>& data) {
+  NSCAssert(flutter_engine_, @"Flutter engine should not be deallocated");
+  [flutter_engine_ dispatchSemanticsAction:action toTarget:target withData:data];
+}
+
+std::unique_ptr<FlutterPlatformNodeDelegate>
+AccessibilityBridgeMacDelegate::CreateFlutterPlatformNodeDelegate() {
+  return std::make_unique<FlutterPlatformNodeDelegateMac>(flutter_engine_);
+}
+
+// Private method
+void AccessibilityBridgeMacDelegate::FireNativeMacNotification(
+    gfx::NativeViewAccessible native_node,
+    NSAccessibilityNotificationName mac_notification) {
+  NSCAssert(mac_notification, @"The notification must not be null.");
+  NSCAssert(native_node, @"The notification target must not be null.");
+  NSAccessibilityPostNotification(native_node, mac_notification);
+}
+
+void AccessibilityBridgeMacDelegate::FireNativeMacNotificationWithUserInfo(
+    gfx::NativeViewAccessible native_node,
+    NSAccessibilityNotificationName mac_notification,
+    NSDictionary* user_info) {
+  NSCAssert(mac_notification, @"The notification must not be null.");
+  NSCAssert(native_node, @"The notification target must not be null.");
+  NSCAssert(user_info, @"The notification data must not be null.");
+  NSAccessibilityPostNotificationWithUserInfo(native_node, mac_notification, user_info);
+}
+
+bool AccessibilityBridgeMacDelegate::IsInGeneratedEventBatch(
+    ui::AXEventGenerator::Event event_type) const {
+  NSCAssert(flutter_engine_, @"Flutter engine should not be deallocated");
+  auto bridge = flutter_engine_.accessibilityBridge.lock();
+  NSCAssert(bridge, @"Accessibility bridge in flutter engine must not be null.");
+  std::vector<ui::AXEventGenerator::TargetedEvent> events = bridge->GetCurrentEvents();
```

Comment:
nit: consider inlining `bridge->GetCurrentEvents()` into the `for` below, since this local is never accessed directly and the expression is short.